monai/networks/blocks/patchembedding.py

```diff
@@ -123,6 +123,7 @@ def __init__(
             with torch.no_grad():
                 pos_embeddings = build_sincos_position_embedding(grid_size, hidden_size, spatial_dims)
```

Comment:
i know this isn't part of the PR but wondering why we don't just do
`self.position_embeddings = build_sincos_position_embedding(grid_size, hidden_size, spatial_dims).float()
` here. @ericspod  any ideas?

Reply:
Agreed, looks like we can directly using the output from `build_sincos_position_embedding`.
https://github.com/Project-MONAI/MONAI/blob/c3a7383494b35f82bc3193797a3b07629e94c0c0/monai/networks/blocks/pos_embed_utils.py#L104

Reply:
Yes, that would work, I can commit a more concise and efficient version of the positional encoding creation if ever, the idea of this PR in the first instance was simply to have the desired behavior (a fixed encoding) with a small fix.

Reply:
Setting `requires_grad = False` is the key thing here, being created within a no-grad context won't set it to False. Within the context there won't be a computation of gradients but we wouldn't have this anyway.

Reply:
Yes, exactly, but here we're also trying to set the positional encoding `requires_grad=False` in `monai/networks/blocks/pos_embed_utils.py`in the code l103/l104

```python
pos_embed = nn.Parameter(pos_emb)
pos_embed.requires_grad = False
```

But the requires grad attributes is not maintained with the torch `copy_` function